src/Templates/AbstractPageTemplate.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * AbstractPageTemplate class.
+ *
+ * Shared logic for page templates.
+ *
+ * @internal
+ */
+abstract class AbstractPageTemplate {
+	/**
+	 * Page Template functionality is only initialized when using a block theme.
+	 */
+	public function __construct() {
+		if ( wc_current_theme_is_fse_theme() ) {
+			$this->init();
+		}
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	protected function init() {
+		add_filter( 'page_template_hierarchy', array( $this, 'page_template_hierarchy' ), 1 );
+		add_filter( 'frontpage_template_hierarchy', array( $this, 'page_template_hierarchy' ), 1 );
```

Comment:
Need to check if this is still needed. It was added when we messed with using template rewrite rules. Unless its needed for order received?